### DISCUSSION TITLE
Vercel caching issue?

### DISCUSSION THREAD

**Discussion by goddrummerofdw** [vercel/community#3259] [Help]
### Summary

*Using Next.js v13.4.3 and deployed in Vercel

For context: I have a login page that utilizes accestokens and refreshtokens that grant the user access to a dashboard.  I'm having issues in staging and prod only. Looks like the dashboard page is being cached which could be the source of my issue.

In the response header, I'm expecting the code below to change the Cach-Control to no-store (Which it does) but I'm also expecting it to change X-Vercel-Cache from HIT to MISS so that it stops caching that page. Am i wrong in thinking this? 
i also tried to add  no-store value in the fetch request(example below)  

next.config.js
`    generateEtags: false,
    headers: () => [
        {
            source: '/dashboard',
            headers: [
                {
                    key: 'Cache-Control',
                    value: 'no-store',
                },
            ],
        },
    ],`
<img width="415" alt="Screenshot 2023-07-15 at 6 07 33 PM" src="https://github.com/vercel/community/assets/73509162/7f5da08a-33cf-4472-a052-e290da80a68a">
<img width="540" alt="Screenshot 2023-07-15 at 6 09 44 PM" src="https://github.com/vercel/community/assets/73509162/7e476185-a783-4cba-ade3-fa23c13da0c4">

### Steps to Reproduce

create an empty page using next.js 13.4.3 then deploy using vercel. Then use the link below try to get a 200 instead of a 304 on refresh.

Note: that hard refresh fixes my issue in staging and prod.

https://nextjs.org/docs/app/api-reference/next-config-js/headers

**Comment by leerob** [comment]
> I'm having issues in staging and prod only

Could you run locally with `next build && next start` to verify?

You shouldn't need to use `headers` if it's just to change caching, I would remove that. If you want `no-store`, you would pass it to the `fetch` call itself, which you are doing. Note that this is caching the _fetch_ and not the entire page.

https://nextjs.org/docs/app/building-your-application/data-fetching/caching

**Reply by goddrummerofdw** [reply]
i went ahead and removed the header from my next.config.js file. Seems to work fine when i use yarn build + yarn start. It still doesn't work when deployed though. 

Is there a way to disable cache on the entire page?

Build: 
<img width="856" alt="Screenshot 2023-07-15 at 8 03 58 PM" src="https://github.com/vercel/community/assets/73509162/69b510fe-43b7-4bfb-8587-152e833a8a0f">

Chrome Inspector Network Ta
<img width="769" alt="Screenshot 2023-07-15 at 8 07 38 PM" src="https://github.com/vercel/community/assets/73509162/51048555-ff9c-43d7-a160-8c12dd3e7ba5">
b: 
<img width="769" alt="Screenshot 2023-07-15 at 8 07 17 PM" src="https://github.com/vercel/community/assets/73509162/f4a52e20-7a1a-4663-a075-2cb0b060018a">

**Reply by iraunit** [reply]
Have you got the solution?

**Reply by goddrummerofdw** [reply]
Nope :/
